### PR TITLE
Minor fixes for Random.choices, tests and docs

### LINKED ISSUE #179: `Random.choices()` division by zero
**Goost and Godot version:**
4cfacd390a0e2f3c832f19677edbb195a234971f, 3.x

**Steps to reproduce:**
This is input from the fuzzer: https://github.com/goostengine/goost-fuzzer/runs/5261841226
```
/home/runner/work/goost-fuzzer/goost-fuzzer/goost/core/math/random.cpp:136:32: runtime error: division by zero
```
```gdscript
Random.choices({"roman" : 22, 22 : 25, BoxShape.new() : BoxShape.new()}, 37, PoolIntArray([]), true)
```

CC @sumadithya

### PR DESCRIPTION
**Pull request by rusty-tendrils**
Closes https://github.com/goostengine/goost/issues/179

### CODE REVIEW

File: core/math/random.cpp [hunk 1]
```diff
@@ -158,20 +158,24 @@ Array Random::choices(const Variant &p_from, int p_count, const PoolIntArray &p_
...
-				} else {
-					sum += weights[i];
-					cumulative_weights.push_back(sum);
-				}
+				ERR_FAIL_COND_V_MSG((weights[i] < 0), Array(), "Weights must be non-negative integers.");
+				sum += weights[i];
+				cumulative_weights.push_back(sum);
 			}
 		}
+		ERR_FAIL_COND_V_MSG((sum == 0), Array(), "Sum of weights cannot be zero");
```
Comment (Xrayez) [inline core/math/random.cpp]: Note that wrapping an expression is not required in error macros:
```suggestion
		ERR_FAIL_COND_V_MSG(sum == 0, Array(), "Sum of weights cannot be zero");
```
This is not required to fix, but just for the future.

File: doc/Random.xml [hunk 2]
```diff
@@ -33,6 +33,9 @@
 			<argument index="2" name="weights" type="PoolIntArray" default="null" />
 			<argument index="3" name="cumulative" type="bool" default="false" />
 			<description>
+				Returns an [Array] of size [code]count[/code] of random elements picked from [code]from[/code] using weights from either [code]weights[/code] or values of [code]from[/code] if [code]from[/code] is a [Dictionary] (and if [code]weights[/codes] is empty).
```
Comment (Xrayez) [inline doc/Random.xml]:
```suggestion
				Returns an [Array] of randomly picked elements from a [code]sequence[/code], with the number of elements equal to [code]count[/code]. The elements are picked according to integer [code]weights[/code] or an array of values from the [code]sequence[/code] if it's a [Dictionary] and if [code]weights[/codes] is empty.
```

I'll rename `from` to `sequence` myself (requires updating bindings). But you could do it yourself as well.

File: doc/Random.xml [hunk 3]
```diff
@@ -33,6 +33,9 @@
...
+				All elements of [code]weights[/code] must be non-negative integers (non-decreasing if cumulative) and of them should not be zero.
```
Comment (Xrayez) [inline doc/Random.xml]: Perhaps "positive" could already imply "non-zero"?

```suggestion
				All elements of [code]weights[/code] must be positive integers, and must not contain zero values. The order of integers should be non-decreasing if [code]cumulative[/code] is [code]true[/code].
```

File: doc/Random.xml [hunk 4]
```diff
@@ -33,6 +33,9 @@
...
+				[code]weights[/code] should of the same size as [code]from[/code]. (if [code]weights[/code] is not empty and if [code]from[/code] is not a [Dictionary])
```
Comment (Xrayez) [inline doc/Random.xml]:
```suggestion
				If [code]weights[/code] is not empty and if [code]sequence[/code] is not a [Dictionary], then the size of [code]weights[/code] must be equal to the size of [code]sequence[/code]. 
```

**Review by Xrayez — COMMENTED** [review]
Mostly documentation suggestions.

File: doc/Random.xml (hunk 3 above)
Comment (rusty-tendrils) [reply doc/Random.xml]: No, by 'non-negative' I meant positive and zero.
The array could have some elements as zero but not all. (I missed writing 'all')

**Review by Xrayez — APPROVED** [review]

**Comment by Xrayez** [comment]
Thanks!

**Pull request merged**